### PR TITLE
[NUI] Adding VertexBuffer to replace PropertyBuffer

### PR DESCRIPTION
**Pull request by DavidIanSteele**
### Description of Change ###

Native PropertyBuffer has been renamed to VertexBuffer.
Creating a new VertexBuffer class to mirror native side.

Has a SetData method to take an array of structs, and create an un-managed buffer before passing it to native side.

(Unlike PropertyBuffer, which takes a marshalled IntPtr to an un-managed buffer)

### API Changes ###

Added:
NUI.VertexBuffer
NUI.Geometry.AddVertexBuffer(VertexBuffer);

### CODE REVIEW

File: src/Tizen.NUI/src/public/Geometry.cs [hunk 1]
```diff
@@ -96,6 +96,19 @@ public uint AddVertexBuffer(PropertyBuffer vertexBuffer)
             return ret;
         }
 
+        /// <summary>
+        /// Adds a VertexBuffer to be used as source of geometry vertices.
+        /// </summary>
+        /// <param name="vertexBuffer">VertexBuffer to be used as source of geometry vertices.</param>
+        /// <returns>Index of the newly added buffer.</returns>
+        /// <since_tizen> 6 </since_tizen>
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/Geometry.cs:104]:
```suggestion
        /// <since_tizen> 8 </since_tizen>
```

File: src/Tizen.NUI/src/public/Geometry.cs [hunk 2]
```diff
@@ -96,6 +96,19 @@ public uint AddVertexBuffer(PropertyBuffer vertexBuffer)
...
+        public uint AddVertexBuffer(VertexBuffer vertexBuffer)
+        {
+            uint ret = Interop.Geometry.Geometry_AddVertexBuffer(swigCPtr, VertexBuffer.getCPtr(vertexBuffer));
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/Geometry.cs:108]: Please don't write if-statement which throws an exception in a single line.

File: src/Tizen.NUI/src/public/Geometry.cs [hunk 3]
```diff
@@ -96,6 +96,19 @@ public uint AddVertexBuffer(PropertyBuffer vertexBuffer)
...
+            return ret;
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/Geometry.cs:109]: Is this return value really necessary?

File: src/Tizen.NUI/src/public/VertexBuffer.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+namespace Tizen.NUI
+{
+    /// <summary>
+    /// VertexBuffer is a handle to an object that contains a buffer of structured data.<br />
+    /// VertexBuffers can be used to provide data to Geometry objects.
+    /// </summary>
+    /// <since_tizen> 3 </since_tizen>
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/VertexBuffer.cs:28]:
```suggestion
    /// <since_tizen> 8 </since_tizen>
```

**Review by adeelkazmi — APPROVED** [review]

**Comment by WonyoungChoi** [comment]
Please tell me the ACR number of this merged PR

**Comment by m-macias** [comment]
https://code.sec.samsung.net/jira/browse/TCSACR-372

**Pull request merged**